publisher/publisher.go

```diff
@@ -254,11 +254,8 @@ func uploadArtifact(conf config, schema uploadArtifactSchema, arch string, uploa
 		l.Println("Uploading apt")
 		err = uploadApt(conf, schema.Src, upload, arch)
 	}
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return
 }
 
 func uploadArtifacts(conf config, schema uploadArtifactsSchema) error {
```

Comment:
since now the downloaded matrix is arc*sysManager shouldn't the upload loop reflect it as well?

The initial idea was to avoid adding a new section and manage it with more src, es:


 ```
- src: "newrelic-infra_systemd_{version}_{arch}.deb"
  arch:
    - 386
    - amd64
  uploads:
    - type: file
      dest: "{dest_prefix}linux/apt/"

- src: "newrelic-infra_upstart_{version}_{arch}.deb"
  arch:
    - 386
    - amd64
  uploads:
    - type: file
      dest: "{dest_prefix}linux/apt/"

- src: "newrelic-infra_sysv_{version}_{arch}.deb"
  arch:
    - 386
    - amd64
  uploads:
    - type: file
      dest: "{dest_prefix}linux/apt/"

 ```

Reply:
I had same solution in mind. It looks that only for one scenario we added extra complexity.